examples/Entity_extraction_for_long_documents.ipynb

```diff
@@ -0,0 +1,370 @@
+    "In this notebook we'll run through this approach:\n",
+    "- We'll load in a long PDF and pull the text out\n",
+    "- We'll create a prompt to be used to extract key bits of information\n",
+    "- We'll chunk up our document and process each chunk to pull any answers out\n",
+    "- We'll then combine them at the end\n",
+    "- This simple approach will then be extended to three more difficult questions"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setup\n",
+    "\n",
+    "We'll take our PDF, a Formula 1 Financial Regulation document on Power Units, and extract the text from it for entity extraction. We'll use this to try to extract answers that are buried in the content."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install textract"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import textract\n",
+    "import os\n",
+    "import openai\n",
+    "\n",
+    "# Extract the raw text from each PDF using textract\n",
+    "text = textract.process('data/fia_f1_power_unit_financial_regulations_issue_1_-_2022-08-16.pdf', method='pdfminer').decode('utf-8')\n",
+    "clean_text = text.replace(\"  \", \" \").replace(\"\\n\", \"; \").replace(';',' ')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Simple Entity Extraction\n",
+    "\n",
+    "We'll start by extracting key bits of information from chunks of a document.\n",
+    "\n",
+    "We'll accomplish this by:\n",
+    "- Creating a template prompt with our questions and an example of the format it expects\n",
+    "- Create a function to take a chunk of text as input, combine with the prompt and get a response\n",
+    "- Run a script to chunk the text, extract answers and output them for parsing"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Extract key pieces of information from this regulation document. \n",
+      "If a particular piece of information is not present, output \"Not specified\". \n",
+      "When you extract a key piece of information, include the closest page number.\n",
+      "Use the following format:\n",
+      "0. Author\n",
+      "1. What is the amount of the power unit cost cap in USD, GBP and EUR\n",
+      "2. What is the value of External Manufacturing Costs in USD\n",
+      "3. What is the Capital Expenditure Limit in USD\n",
+      "\n",
+      "Document: \"\"\"<document>\"\"\"\n",
+      "\n",
+      "0. Author: Tom Anderson (Page 1)\n",
+      "1.\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Example prompt - \n",
+    "document = '<document>'\n",
+    "template_prompt=f'''Extract key pieces of information from this regulation document. \n",
```

Comment:
can cut the trailing spaces on some of the lines in the templates

Reply:
Updated